src/main/webapp/app/shared/layouts/profiles/profile.service.ts

```diff
@@ -12,31 +12,36 @@ import { Saml2Config } from 'app/home/saml2-login/saml2.config';
 @Injectable({ providedIn: 'root' })
 export class ProfileService {
     private infoUrl = SERVER_API_URL + 'management/info';
-    private profileInfo: BehaviorSubject<ProfileInfo | undefined>;
+    private profileInfo: Subject<ProfileInfo>;
 
     constructor(private http: HttpClient, private featureToggleService: FeatureToggleService) {}
 
-    getProfileInfo(): BehaviorSubject<ProfileInfo | undefined> {
+    getProfileInfo(): Subject<ProfileInfo> {
```

Comment:
```suggestion
    getProfileInfo(): Observable<ProfileInfo> {
```

Reply:
@Olimbia23 thank you a lot for your continuous support. 

Could you please explain to me why we should use `Observable` here?
Moreover, what does the `OperatorFunction<ProfileInfo | undefined, ProfileInfo>` do exactly?
Can this change, similar to the change to `Subject`, influence other files? (See `about-us.component.spec` where I had to change `BehaviorSubject` to `Observable`. The tests still work. Or is this change safe?)

Thank you in advance!

Reply:
1. Why `Observable`?
Because the `pipe` method returns an `Observable<T>`

2. What does `OperatorFunction<ProfileInfo | undefined, ProfileInfo>`?
Without this cast operator, the return type would be either `Observable<ProfileInfo | undefined>`, but since our method `getProfileInfo()` should return only data types of `Observable<ProfileInfo>`, then we need to use the casting operator. Our Filter method makes sure that we are not returning any undefined data.

3. Can this influence other files?
Yes, because the API is changed (returned type of `getProfileInfo` function). 
Before it was returning `BehaviorSubject` where one can push new elements into by calling the `next` function. `Observable`'s do not have the next function and therefore can not be mutated.
However, it seems like all usages of this function (outside of tests) are just subscribing to the the `Observable` stream without writing anything into it via the `next` method, which is only available for `BehaviorSubject`. Therefore this is not a breaking change (outside of tests, which have to be adjusted accordingly)

Reply:
Thank you once more for your great help and detailed explanations!